### PR TITLE
feat(platform-browser): add token marking which the type of animation…

### PR DESCRIPTION
**Pull request by josephperrott**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently, it is not possible to determine if a component is running in a `BrowserAnimationsModule` or `NoopAnimationsModule` context at runtime.
Issue Number: N/A

## What is the new behavior?
An token is provided as a marker to be able to determine which animation context is being used for the component at runtime.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/platform-browser/animations/src/providers.ts [hunk 1]
```diff
@@ -50,12 +64,16 @@ const SHARED_ANIMATION_PROVIDERS: Provider[] = [
  */
 export const BROWSER_ANIMATIONS_PROVIDERS: Provider[] = [
   {provide: AnimationDriver, useFactory: instantiateSupportedAnimationDriver},
+  {provide: BROWSER_ANIMATIONS_MODULE, useValue: BrowserAnimationsModules.BrowserAnimations},
```
Comment (matsko) [inline packages/platform-browser/animations/src/providers.ts]: `ANIMATION_MODULE_TYPE`

File: packages/platform-browser/animations/src/providers.ts [hunk 2]
```diff
@@ -50,12 +64,16 @@ const SHARED_ANIMATION_PROVIDERS: Provider[] = [
...
   ...SHARED_ANIMATION_PROVIDERS
 ];
 
 /**
  * Separate providers from the actual module so that we can do a local modification in Google3 to
  * include them in the BrowserTestingModule.
  */
-export const BROWSER_NOOP_ANIMATIONS_PROVIDERS: Provider[] =
-    [{provide: AnimationDriver, useClass: NoopAnimationDriver}, ...SHARED_ANIMATION_PROVIDERS];
+export const BROWSER_NOOP_ANIMATIONS_PROVIDERS: Provider[] = [
```
Comment (matsko) [inline packages/platform-browser/animations/src/providers.ts]: Could we make a test inside of https://github.com/angular/angular/blob/master/packages/core/test/animation/animation_integration_spec.ts

This way if the token is removed then we'll know if something fails in the project.

```ts
 // put this at the top of the file just under the top-level describe() call...
 describe('animation modules', function() {
    it('should hint at BrowserAnimationsModule being used', () => {
      TestBed.configureTestingModule({
        declarations: [ SharedAnimationCmp ],
        imports: [BrowserAnimationsModule]
      });
     
     TestBed.configureTestingModule({declarations: [SharedAnimationCmp]});
      const fixture = TestBed.createComponent(Cmp);
      const cmp = fixture.componentInstance;
      expect(cmp.tokenValue).toEqual(...);
    });

    it('should hint at NoopAnimationsModule being used', () => {
      TestBed.configureTestingModule({
        declarations: [ SharedAnimationCmp ],
        imports: [NoopAnimationsModule]
      });
     
     TestBed.configureTestingModule({declarations: [SharedAnimationCmp]});
      const fixture = TestBed.createComponent(Cmp);
      const cmp = fixture.componentInstance;
      expect(cmp.tokenValue).toEqual(...);
    });
  });

class SharedAnimationCmp {
  public tokenVaue;
  constructor( /* inject the token */ ) {}
} 
```

**Review by matsko — CHANGES_REQUESTED** [review]
Please fix the things I pointed out.

**Review by matsko — APPROVED** [review]

File: tools/public_api_guard/platform-browser/animations.d.ts [hunk 3]
```diff
@@ -1,3 +1,12 @@
+/** @experimental */
+export declare const ANIMATION_MODULE_TYPE: InjectionToken<AnimationModuleType>;
```
Comment (IgorMinar) [inline tools/public_api_guard/platform-browser/animations.d.ts]: I discussed this with @matsko.

Let's change this to `ANIMATION_MODULE_NAME: InjectionToken<"BrowserAnimationsModule"|"NoopAnimationsModule">` 

It's simpler and exposes less of a API surface.

**Review by IgorMinar — CHANGES_REQUESTED** [review]

**Review by matsko — APPROVED** [review]

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**